pkgs/tools/admin/ansible/default.nix

```diff
@@ -0,0 +1,82 @@
+{ stdenv, fetchurl, fetchFromGitHub, python3
+, windowsSupport ? false
+}:
+
+let
+  oldJinja = python3.override {
+    packageOverrides = self: super: {
+      jinja2 = super.jinja2.overridePythonAttrs (oldAttrs: rec {
+        version = "2.8.1";
+        src = oldAttrs.src.override {
+          inherit version;
+          sha256 = "14aqmhkc9rw5w0v311jhixdm6ym8vsm29dhyxyrjfqxljwx1yd1m";
+        };
+        doCheck = false;
+      });
+    };
+  };
+
+  generic = { version, sha256, py ? python3 }: py.pkgs.buildPythonPackage rec {
+    pname = "ansible";
+    inherit version;
+
+    src = fetchurl {
+      url = "http://releases.ansible.com/ansible/${pname}-${version}.tar.gz";
+      inherit sha256;
+    };
+
+    prePatch = ''
+      sed -i "s,/usr/,$out," lib/ansible/constants.py
+    '';
+
+    doCheck = false;
+    dontStrip = true;
+    dontPatchELF = true;
+    dontPatchShebangs = false;
+
+    propagatedBuildInputs = with py.pkgs; [
+      pycrypto paramiko jinja2 pyyaml httplib2 boto six netaddr dnspython
+    ] ++ stdenv.lib.optional windowsSupport pywinrm;
+
+    meta = with stdenv.lib; {
+      homepage = http://www.ansible.com;
+      description = "A simple automation tool";
+      license = with licenses; [ gpl3 ] ;
+      maintainers = with maintainers; [ jgeerds joamaki ];
+      platforms = with platforms; linux ++ darwin;
+    };
+  };
+
+in rec {
+  # I really don't know why we carry so many versions...
```

Comment:
We should only have the official supported releases here, which is 2.2 up to 2.4 https://docs.ansible.com/ansible/2.4/release_and_maintenance.html#release-status

Reply:
Done